Assets/Project/Modules/Enemies/Turret/Scripts/scripts/TurretShooting.cs

```diff
@@ -0,0 +1,97 @@
+       [SerializeField] private Transform _firePoint;
+       private float timer = 0;
+       private ParabolicProjectile _currentProjectile;
+       
+        
+       [SerializeField] private float _squashAmountY = 2.6f;
+       [SerializeField] private float _squashAmountXZ = 2.6f;
+       [SerializeField] private float _stretchAmountY = 2.6f;
+       [SerializeField] private float _stretchAmountXZ = 2.8f;
+       [SerializeField] private float _squashAndStretchTime = 0.5f;
+
+       private bool animationOn = false;
+       [SerializeField] private float _playerDistanceThreshold;
+       private float _squaredPlayerDistanceThreshold;
+
+        public void Configure(AEnemyMediator turetMediator,Core.Pool.ObjectPool projectilePool,Transform transform)
+        {
+            _mediator = turetMediator;
+            _projectilePool = projectilePool;
+            _playerTransform = transform;
+            _currentProjectile = _projectilePool.Spawn<ParabolicProjectile>(_firePoint.position, quaternion.identity);
+            _currentProjectile.PrepareShot(_firePoint,_playerTransform);
+            _squaredPlayerDistanceThreshold = _playerDistanceThreshold * _playerDistanceThreshold;
+        }
+
+        private void Update()
+        {
+            if (IsPlayerAtCloseDistance())
+            {
+                if (timer >= timeBetweenShots - 1)
+                {
+                    _currentProjectile.StartAiming();
+
+                    if (!animationOn)
+                    {
+                        SquashAndStretch();
+                    }
+
+                    if (timer >= timeBetweenShots)
+                    {
+                        _currentProjectile.Shoot();
+                        timer = 0;
+                        _currentProjectile =
+                            _projectilePool.Spawn<ParabolicProjectile>(_firePoint.position, quaternion.identity);
+                        _currentProjectile.PrepareShot(_firePoint, _playerTransform);
+                    }
+                }
+
+                timer += Time.deltaTime;
+            }
+            else
+            {
+                timer = 0;
+            }
+        }
+
+        private float GetPlayerSqrMagnitude()
+        {
+            return (_playerTransform.position - transform.position).sqrMagnitude;
+        }
+        
+        private bool IsPlayerAtCloseDistance()
+        {
+            return GetPlayerSqrMagnitude() < _squaredPlayerDistanceThreshold;
+        }
+        private async UniTaskVoid SquashAndStretch()
+        {
+            animationOn = true;
+            // Squash
+            await transform.DOScale(new Vector3(_squashAmountXZ, _squashAmountY, _squashAmountXZ), 0.9f)
+                .AsyncWaitForCompletion();
+            //Stretch
+            await transform.DOScale(new Vector3(_stretchAmountXZ, _stretchAmountY, _stretchAmountXZ), 0.2f)
+                .AsyncWaitForCompletion();
+            animationOn = false;
+
+        }
```

Comment:
Shouldn't this class be using the "squash and stretch" script that you created a while ago, instead of this?

Reply:
It was a provisional class, now I have renamed it to Slime Animator Controller and basically it has functions where animations need to be triggered and particles spawned.